tests/instruments/test_instrument.py

```diff
@@ -121,30 +120,61 @@ def test_read(self, instr):
         instr.read()
         assert instr.adapter.method_calls == [mock.call.read()]
 
+    def test_write_bytes(self, instr):
+        instr.write_bytes(b"abc")
+        assert instr.adapter.method_calls == [mock.call.write_bytes(b"abc")]
+
+    def test_read_bytes(self, instr):
+        instr.read_bytes(5)
+        assert instr.adapter.method_calls == [mock.call.read_bytes(5)]
+
+
+class TestWaiting:
+    @pytest.fixture()
+    def instr(self):
+        class Faked(Instrument):
+            def wait_till_read(self, query_delay=0):
+                self.waited = query_delay
+        return Faked(ProtocolAdapter(), name="faked")
+
     def test_ask(self, instr):
+        instr.adapter.comm_pairs = [("abc", "resp")]
         instr.ask("abc")
-        assert instr.adapter.method_calls == [mock.call.ask('abc')]
+        assert instr.waited == 0
 
-    def test_values(self, instr):
-        instr.values("abc")
-        assert instr.adapter.method_calls == [mock.call.values('abc')]
+    def test_ask_time(self, instr):
+        instr.adapter.comm_pairs = [("abc", "resp")]
+        instr.ask("abc", query_delay=10)
+        assert instr.waited == 10
 
     def test_binary_values(self, instr):
-        instr.binary_values("abc", dtype=int)
-        assert instr.adapter.method_calls == [mock.call.binary_values('abc', 0, int)]
+        instr.adapter.comm_pairs = [("abc", "abcdefgh")]
+        instr.binary_values("abc")
+        assert instr.waited == 0
```

Comment:
I'm not sure why all these tests reduce to only asserting the waited time, instead of the method calls as before? Is the diff view confusing me?

Reply:
The diff is confusing:
I deleted the tests, that `ask` and `values` call the Adapter (which they do not do anymore, they call Instrument methods instead).
Additionally I added test, which verify, that these methods do call the wait method. For this, asserting the wait time is enough

Reply:
OK, thanks!